src/components/Icon/Icon.tsx

```diff
@@ -15,12 +16,49 @@ const COLORS_WITH_BACKDROPS = [
   'inkLighter',
 ];
 
+const NEW_DESIGN_LANGUAGE_COLORS = [
+  'base',
+  'disabled',
+  'hovered',
+  'pressed',
+  'subdued',
+  'critical',
+  'warning',
+  'highlight',
+  'interactive',
+  'success',
+  'primary',
+  'primaryDisabled',
+  'primaryHovered',
+  'primaryPressed',
+];
```

Comment:
This repeats the type added as part of this, because I want to be able to iterate through these as values. If anyone knows a DRYer way to accomplish this, I'm keen to hear it.

Reply:
This will do the trick:

```ts
const NEW_DESIGN_LANGUAGE_COLORS = [
  'base',
  'disabled',
  'hovered',
  'pressed',
  'subdued',
  'critical',
  'warning',
  'highlight',
  'success',
  'primary',
  'primaryDisabled',
  'primaryHovered',
  'primaryPressed',
] as const;  // <-- const context
type NewDesignLanguageColor = typeof NEW_DESIGN_LANGUAGE_COLORS[number];
```

The new [const contexts](https://github.com/Microsoft/TypeScript/pull/29510) (supported as of v3.4) sets the type on the array as a readonly array of the string literal values, instead of an array of strings. So then getting the types of each value with a key of type `number` results in a string literal type. Although using `any` produces the same result.